File: scramjet/packages/demo/scratch/merge-onefile.mjs
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const root = path.dirname(fileURLToPath(import.meta.url));
const htmlPath = path.join(root, '1.html');
const gamedataPath = path.join(root, '..', 'gamedata.js');
const bundlePath = path.join(root, 'quantum-bundle.js');

let html = fs.readFileSync(htmlPath, 'utf8');
let gd = fs.readFileSync(gamedataPath, 'utf8');
const cut = gd.indexOf('games.sort');
if (cut > 0) gd = gd.slice(0, cut);
gd = gd.replace('const games', 'window.games', 1).trim() + '\n';

const bundle = fs.existsSync(bundlePath) ? fs.readFileSync(bundlePath, 'utf8') : '';
const inject = `<script>\n${gd}\n</script>\n<script>\n${bundle}\n</script>\n`;

const lucideTag = '<script src="https://unpkg.com/lucide@latest/dist/umd/lucide.js"></script>';
if (html.includes('window.games =') && html.includes('QuantumExtras')) {
  console.log('Already merged — skipping duplicate inject');
  process.exit(0);
}
if (!html.includes(lucideTag)) {
  console.error('Lucide script tag not found');
  process.exit(1);
}
html = html.replace(lucideTag, inject + lucideTag);
fs.writeFileSync(htmlPath, html, 'utf8');
console.log('Merged into 1.html');

File: scramjet/packages/demo/scratch/merge-onefile.py
# Merge gamedata + bundle into 1.html (single file)
import re
from pathlib import Path

root = Path(__file__).parent
html_path = root / "1.html"
gamedata_path = root.parent / "gamedata.js"
bundle_path = root / "quantum-bundle.js"

html = html_path.read_text(encoding="utf-8")
gd = gamedata_path.read_text(encoding="utf-8")
# Only games array (no renderItems at end)
cut = gd.find("games.sort")
if cut > 0:
    gd = gd[:cut]
gd = gd.replace("const games", "window.games", 1).strip() + "\n"

bundle = bundle_path.read_text(encoding="utf-8") if bundle_path.exists() else ""

inject = f'<script>\n{gd}\n</script>\n<script>\n{bundle}\n</script>\n'

marker = "<!-- ======================== -->\n<!-- LUCIDE ICONS CDN SCRIPT  -->"
if marker not in html:
    marker = '<script src="https://unpkg.com/lucide@latest/dist/umd/lucide.js"></script>'
    html = html.replace(marker, inject + marker, 1)
else:
    html = html.replace(marker, inject + marker, 1)

# Scramjet scripts after </style> if missing
if "scramjet.js" not in html:
    html = html.replace(
        "</style>",
        '</style>\n<script src="https://intelligent-dedication-production-f8d8.up.railway.app/scramjet/scramjet.js"></script>\n<script src="https://intelligent-dedication-production-f8d8.up.railway.app/controller/controller.api.js"></script>',
        1,
    )

html_path.write_text(html, encoding="utf-8")
print("Merged into 1.html")
